### PR TITLE
Add more debugging info to udev builtin net_id

### PR DESCRIPTION
**Pull request by keszybz**
Prompted by the recent question on the mailing list. Even with debugging enabled, it was very hard to figure out why the names that were chosen were chosen.

### CODE REVIEW

File: src/udev/udev-builtin-net_id.c [hunk 1]
```diff
@@ -373,20 +394,27 @@ static int dev_pci_slot(sd_device *dev, const LinkInfo *info, NetNames *names) {
...
         if (!snprintf_ok(slots, sizeof slots, "%s/slots", syspath))
-                return -ENAMETOOLONG;
+                return log_device_debug_errno(dev, SYNTHETIC_ERRNO(ENAMETOOLONG),
+                                              "Cannot access %s/slots: %m", syspath);
 
         dir = opendir(slots);
         if (!dir)
-                return -errno;
+                return log_device_debug_errno(dev, errno,
+                                              "Cannot access %s/slots: %m", syspath);
```
Comment (bluca) [inline src/udev/udev-builtin-net_id.c]: `slots` instead of `syspath`?

**Review by bluca — APPROVED** [review]

File: src/udev/udev-builtin-net_id.c [hunk 2]
```diff
@@ -210,8 +216,13 @@ static int dev_pci_onboard(sd_device *dev, const LinkInfo *info, NetNames *names
                 l = strpcpyf(&s, l, "d%lu", dev_port);
         if (l == 0)
                 names->pci_onboard[0] = '\0';
+        log_device_debug(dev, "Onboard index identifier: index=%lu phys_port=%s dev_port=%lu → %s",
+                         idx, strempty(info->phys_port_name), dev_port,
+                         empty_to_na(names->pci_slot));
 
-        if (sd_device_get_sysattr_value(names->pcidev, "label", &names->pci_onboard_label) < 0)
+        if (sd_device_get_sysattr_value(names->pcidev, "label", &names->pci_onboard_label) >= 0)
+                log_device_debug(dev, "Onboard label: %s", names->pci_onboard_label);
```
Comment (yuwata) [inline src/udev/udev-builtin-net_id.c]: `dev` -> `names->pcidev`?

File: src/udev/udev-builtin-net_id.c [hunk 3]
```diff
@@ -327,9 +341,12 @@ static int dev_pci_slot(sd_device *dev, const LinkInfo *info, NetNames *names) {
 
         r = sd_device_get_sysname(names->pcidev, &sysname);
         if (r < 0)
-                return r;
+                return log_device_debug_errno(names->pcidev, r, "Failed to get sysname: %m");
 
-        if (sscanf(sysname, "%x:%x:%x.%u", &domain, &bus, &slot, &func) != 4)
+        r = sscanf(sysname, "%x:%x:%x.%u", &domain, &bus, &slot, &func);
+        log_device_debug(dev, "Parsing slot information from sysname \"%s\": %s",
```
Comment (yuwata) [inline src/udev/udev-builtin-net_id.c]: Also here

File: src/udev/udev-builtin-net_id.c [hunk 4]
```diff
@@ -373,20 +394,26 @@ static int dev_pci_slot(sd_device *dev, const LinkInfo *info, NetNames *names) {
...
 
+        log_device_debug(dev, "PCI path identifier: domain=%u bus=%u slot=%u func=%u phys_port=%s dev_port=%lu → %s",
+                         domain, bus, slot, func, strempty(info->phys_port_name), dev_port,
+                         empty_to_na(names->pci_path));
+
         /* ACPI _SUN — slot user number */
         r = sd_device_new_from_subsystem_sysname(&pci, "subsystem", "pci");
         if (r < 0)
-                return r;
+                return log_debug_errno(r, "sd_device_new_from_subsystem_sysname failed: %m");
```
Comment (yuwata) [inline src/udev/udev-builtin-net_id.c:404]: Maybe better to add "()" to the function name?

File: src/udev/udev-builtin-net_id.c [hunk 5]
```diff
@@ -373,20 +394,26 @@ static int dev_pci_slot(sd_device *dev, const LinkInfo *info, NetNames *names) {
...
 
         r = sd_device_get_syspath(pci, &syspath);
         if (r < 0)
-                return r;
+                return log_device_debug_errno(pci, r, "sd_device_get_syspath failed: %m");
+
         if (!snprintf_ok(slots, sizeof slots, "%s/slots", syspath))
-                return -ENAMETOOLONG;
+                return log_device_debug_errno(dev, SYNTHETIC_ERRNO(ENAMETOOLONG),
```
Comment (yuwata) [inline src/udev/udev-builtin-net_id.c:411]: dev -> pci

File: src/udev/udev-builtin-net_id.c [hunk 6]
```diff
@@ -373,20 +394,26 @@ static int dev_pci_slot(sd_device *dev, const LinkInfo *info, NetNames *names) {
...
+                                              "Cannot access %s/slots: %m", syspath);
 
         dir = opendir(slots);
         if (!dir)
-                return -errno;
+                return log_device_debug_errno(dev, errno, "Cannot access %s: %m", slots);
```
Comment (yuwata) [inline src/udev/udev-builtin-net_id.c:416]: Also here.

File: src/udev/udev-builtin-net_id.c [hunk 7]
```diff
@@ -262,7 +273,10 @@ static bool is_pci_bridge(sd_device *dev) {
                 return false;
 
         /* PCI device subclass 04 corresponds to PCI bridge */
-        return strneq(p + 2, "04", 2);
+        bool b = strneq(p + 2, "04", 2);
+        if (b)
+                log_device_debug(dev, "Device is a bridge.");
```
Comment (yuwata) [inline src/udev/udev-builtin-net_id.c]: bridge -> PCI bridge

File: src/udev/udev-builtin-net_id.c [hunk 8]
```diff
@@ -426,8 +453,10 @@ static int dev_pci_slot(sd_device *dev, const LinkInfo *info, NetNames *names) {
                                  * slot index if the device is a PCI bridge, because it can have other child
                                  * devices that will try to claim the same index and that would create name
                                  * collision. */
-                                if (naming_scheme_has(NAMING_BRIDGE_NO_SLOT) && is_pci_bridge(hotplug_slot_dev))
+                                if (naming_scheme_has(NAMING_BRIDGE_NO_SLOT) && is_pci_bridge(hotplug_slot_dev)) {
+                                        log_device_debug(dev, "Not using slot information because device is a PCI bridge.");
```
Comment (yuwata) [inline src/udev/udev-builtin-net_id.c]: dev -> pci

File: src/udev/udev-builtin-net_id.c (hunk 2 above)
Comment (keszybz) [reply src/udev/udev-builtin-net_id.c]: That is on purpose. We want to report the label that pertains to `dev`, even though we extracted it from `pcidev`…

Comment (keszybz) [reply src/udev/udev-builtin-net_id.c]: I added "from PCI device" now, here and in the next one to make this clearer.

File: src/udev/udev-builtin-net_id.c (hunk 6 above)
Comment (keszybz) [reply src/udev/udev-builtin-net_id.c:416]: Changed.

Comment (keszybz) [reply src/udev/udev-builtin-net_id.c:416]: Actually no, I think it's better to use `dev`. The path we couldn't opened is printed, so this should be unambiguous.

File: src/udev/udev-builtin-net_id.c (hunk 8 above)
Comment (keszybz) [reply src/udev/udev-builtin-net_id.c]: I changed the text to "because the PCI device is a bridge".

**Comment by keszybz** [comment]
There are often two devices involved: the device for which we're generating the name, and e.g. the pci device. When we print information about errors, we want to use the specific device name. But when we find information, we should use the first device, so that the debug logs all refer to the same device. I changed the texts a bit, hopefully this is clearer now.

**Comment by yuwata** [comment]
That makes sense. LGTM.

**Review by yuwata — APPROVED** [review]

**Pull request merged**